src/components/OffersList/OffersList.tsx

```diff
@@ -2,12 +2,24 @@ import React, { MouseEventHandler, useContext, useState } from 'react';
 import styles from './OffersList.module.scss';
 import Offer from '../Offer/Offer';
 import { OfferData } from '../../interfaces/OfferData';
+import { DataContext } from '../../context/DataContext';
+import OfferDetailsModal from '../Modal/OfferDetailsModal';
 
 interface OffersListProps {
    offers: OfferData[];
 }
 
 const OffersList: React.FC<OffersListProps> = ({ offers }) => {
+   const { selectedOfferId, setSelectedOfferId, isModalOpen, setIsModalOpen } =
+      useContext(DataContext);
```

Comment:
**nitpick:** I would split contexts as its currently one big DataContext - it's not too meaningful. You could do `ModalContext`, and `JobOffersContext` for example